### PR TITLE
Add `device_name` classmethod in `Accelerator`.

### LINKED ISSUE #17355: Add device information to the accelerator config message
### Description & Motivation

Revamp

```bash
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
```

To

```bash
GPU available: M1, using 1 devices
TPU available: v4-8, using 0 devices
IPU available: False, using 0 devices
HPU available: False, using 0 devices
```

The relevant code is: https://github.com/Lightning-AI/lightning/blob/f14ee9edbc8269054e12daf30b8681d530e73369/src/lightning/pytorch/trainer/setup.py#L145-L171

### Pitch

If the accelerator is available, `True` changes to the actual name of the accelerator used.
If it's unavailable, we still show `False`.

For GPUs, the `cuda|mps` field is gone, as it should be clear from the device.

I also propose that the GPU field shows the number of devices, instead of a used boolean.

We can get this info via
```python
# CUDA
torch.cuda.get_device_name()

# TPU
from torch_xla.experimental import tpu
import torch_xla.core.xla_env_vars as xenv
tpu.get_tpu_env()[xenv.ACCELERATOR_TYPE]
```

For MPS, HPU, IPU we would need to find out if we can get this information. In the meantime, we can still fallback "True" for them.

### Alternatives

One caveat is that this might be misleading with heterogeneous devices, as only rank zero prints this information.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by GdoongMathew**
## What does this PR do?

Fixes #17355

* 1 GPU
```text
GPU available: NVIDIA GeForce RTX 3050 4GB Laptop GPU, using: 1 devices.
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
```
* 2 GPUs (Kaggle)
```python
>>> t = Trainer(devices=2)
INFO: GPU available: Tesla T4, using: 2 devices.
INFO: TPU available: False, using: 0 TPU cores
INFO: HPU available: False, using: 0 HPUs
```
* No cuda GPU available
```text
GPU available: False, using: 0 devices.
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
```
* MPS
```text
GPU available: Apple M1 Pro, using: 1 devices.
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
```
* XLA (Kaggle)
```text
GPU available: False, using: 0 devices.
TPU available: v3-8, using: 8 TPU cores
HPU available: False, using: 0 HPUs
```

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--21112.org.readthedocs.build/en/21112/

### CODE REVIEW

File: src/lightning/pytorch/accelerators/cuda.py [hunk 1]
```diff
@@ -113,6 +113,13 @@ def register_accelerators(cls, accelerator_registry: _AcceleratorRegistry) -> No
             description=cls.__name__,
         )
 
+    @classmethod
+    @override
+    def device_name(cls, device: Optional[_DEVICE] = None) -> str:
+        if not cls.is_available():
+            return "False"
```
Comment (Borda) [inline src/lightning/pytorch/accelerators/cuda.py]: return False as a name does not make a sense, in such case just return empty string

Comment (GdoongMathew) [reply src/lightning/pytorch/accelerators/cuda.py]: Good call, will apply this to mps and xla accelerator as well.

Comment (Borda) [reply src/lightning/pytorch/accelerators/cuda.py]:
```suggestion
            return ""
```

**Comment by GdoongMathew** [comment]
Hi @Borda , not sure how I could ping / DM you in Linkedin, but here's my [linkedin](https://www.linkedin.com/in/%E6%9B%B8%E4%B8%9E-%E8%91%A3-06718b179/?locale=en_US).

**Comment by GdoongMathew** [comment]
Results in Kaggle:
* CPU: https://www.kaggle.com/code/gdoong/lightning-cpu?scriptVersionId=258513715&cellId=2
* GPU x 2: https://www.kaggle.com/code/gdoong/lightning-gpu?scriptVersionId=258514966&cellId=2
* TPU: https://www.kaggle.com/code/gdoong/lightning-tpu?scriptVersionId=258515194&cellId=2

File: src/lightning/pytorch/accelerators/cuda.py (hunk 1 above)
Comment (GdoongMathew) [reply src/lightning/pytorch/accelerators/cuda.py]: fixed, and thanks for the help~

File: src/lightning/pytorch/accelerators/accelerator.py [hunk 2]
```diff
@@ -45,3 +45,8 @@ def get_device_stats(self, device: _DEVICE) -> dict[str, Any]:
 
         """
         raise NotImplementedError
+
+    @classmethod
+    def device_name(cls, device: Optional[_DEVICE] = None) -> str:
```
Comment (justusschock) [inline src/lightning/pytorch/accelerators/accelerator.py:50]: this is not a classmethod. A classmethod is returning an instance of the class. This would be a static method

File: src/lightning/pytorch/accelerators/mps.py [hunk 3]
```diff
@@ -87,6 +87,14 @@ def register_accelerators(cls, accelerator_registry: _AcceleratorRegistry) -> No
             description=cls.__name__,
         )
 
+    @classmethod
```
Comment (justusschock) [inline src/lightning/pytorch/accelerators/mps.py:96]: not a class method (see above)

File: src/lightning/pytorch/accelerators/xla.py [hunk 4]
```diff
@@ -53,3 +54,25 @@ def get_device_stats(self, device: _DEVICE) -> dict[str, Any]:
     @override
     def register_accelerators(cls, accelerator_registry: _AcceleratorRegistry) -> None:
         accelerator_registry.register("tpu", cls, description=cls.__name__)
+
+    @classmethod
```
Comment (justusschock) [inline src/lightning/pytorch/accelerators/xla.py:67]: not a class method (see above)

File: src/lightning/pytorch/accelerators/cuda.py [hunk 5]
```diff
@@ -113,6 +113,13 @@ def register_accelerators(cls, accelerator_registry: _AcceleratorRegistry) -> No
             description=cls.__name__,
         )
 
+    @classmethod
```
Comment (justusschock) [inline src/lightning/pytorch/accelerators/cuda.py:121]: not a class method (see above)

**Review by justusschock — COMMENTED** [review]
How would you handle heterogeneous gpu types? E.g. say I have one 3090 and one 4090 in my workstation. This doesn't handle it at all.

**Comment by GdoongMathew** [comment]
> How would you handle heterogeneous gpu types? E.g. say I have one 3090 and one 4090 in my workstation. This doesn't handle it at all.

Hi @justusschock , based on the [change](https://github.com/GdoongMathew/pytorch-lightning/blob/f124d55ebc24c5e7de99e77533885d6d2778621c/src/lightning/pytorch/trainer/setup.py#L147), the output should look something like
```
GPU available: RTX 3090, RTX 4090, using: 2 devices.
```
that being said, if the setup is more complex - for example 2 x 3090 and 1 x 4090 - the current output might not fully reflect that.

File: src/lightning/pytorch/accelerators/accelerator.py (hunk 2 above)
Comment (GdoongMathew) [reply src/lightning/pytorch/accelerators/accelerator.py:50]: I think it might be better if the method always falls back to returning `True` or `False` when it isn’t implemented in the individual accelerator class. Since it needs to check availability through `cls.is_available()`, it would likely make sense to implement it as a classmethod. 

What do you think? open to other approaches as well~

**Comment by GdoongMathew** [comment]
On second thought, I may have oversimplified the question. It looks like the current implementation could cause issues with the DDP training strategy, since it tries to access a non-existent device ID on rank zero.

Thanks for your review @Borda & @justusschock , I’ll mark this PR as WIP for now. Any further suggestions are welcome~

**Comment by justusschock** [comment]
> that being said, if the setup is more complex - for example 2 x 3090 and 1 x 4090 - the current output might not fully reflect that.

@GdoongMathew I think that's fine. it's just important to reflect all available gpu types as that might impact memory etc.

**Comment by GdoongMathew** [comment]
> On second thought, I may have oversimplified the question. It looks like the current implementation could cause issues with the DDP training strategy, since it tries to access a non-existent device ID on rank zero.
> 
> Thanks for your review @Borda & @justusschock , I’ll mark this PR as WIP for now. Any further suggestions are welcome~

To follow up on my own concern: it seems the device_ids property refers to devices on the current node, not all devices across the world view. So the current implementation probably won’t cause any issues, aside from not being able to list device types from other nodes.

**Comment by GdoongMathew** [comment]
Hi @Borda , a friendly ping for a review ~

**Comment by Borda** [comment]
> Hi @Borda , a friendly ping for a review ~

unfortunatly, I am not in a position to approve a feature type PR, lest ping @ethanwharris
